store/local_cache_supplier_schemes.go

```diff
@@ -41,18 +42,20 @@ func (s *LocalCacheSupplier) SchemeGetByName(ctx context.Context, schemeName str
 }
 
 func (s *LocalCacheSupplier) SchemeDelete(ctx context.Context, schemeId string, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
-	defer s.doInvalidateCacheCluster(s.schemeCache, schemeId)
-	defer s.doClearCacheCluster(s.roleCache)
+	result := s.Next().SchemeDelete(ctx, schemeId, hints...)
+
+	s.doInvalidateCacheCluster(s.schemeCache, schemeId)
+	s.doClearCacheCluster(s.roleCache)
 
-	return s.Next().SchemeDelete(ctx, schemeId, hints...)
+	return result
 }
 
 func (s *LocalCacheSupplier) SchemeGetAllPage(ctx context.Context, scope string, offset int, limit int, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
 	return s.Next().SchemeGetAllPage(ctx, scope, offset, limit, hints...)
 }
 
 func (s *LocalCacheSupplier) SchemePermanentDeleteAll(ctx context.Context, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
-	defer s.doClearCacheCluster(s.schemeCache)
-
-	return s.Next().SchemePermanentDeleteAll(ctx, hints...)
+	result := s.Next().SchemePermanentDeleteAll(ctx, hints...)
+	s.doClearCacheCluster(s.schemeCache)
```

Comment:
Not your bug, but we need to nuke the roles cache here too.

Reply:
Cool, I'll add that as well